cmd/xl-storage.go

```diff
@@ -2367,6 +2366,41 @@ func (s *xlStorage) AppendFile(ctx context.Context, volume string, path string,
 	return nil
 }
 
+// checkPart is a light check of an existing and size of a part, without doing a bitrot operation
+// For any unexpected error, return checkPartUnknown (zero)
+func (s *xlStorage) checkPart(volumeDir, path, dataDir string, partNum int, expectedSize int64, skipAccessCheck bool) (resp int) {
+	partPath := pathJoin(path, dataDir, fmt.Sprintf("part.%d", partNum))
+	filePath := pathJoin(volumeDir, partPath)
+	st, err := Lstat(filePath)
+	if err != nil {
+		if osIsNotExist(err) {
+			if !skipAccessCheck {
+				// Stat a volume entry.
+				if verr := Access(volumeDir); verr != nil {
+					if osIsNotExist(verr) {
+						resp = checkPartVolumeNotFound
+					}
+					return
+				}
+			}
+		}
+		if osErrToFileErr(err) == errFileNotFound {
+			resp = checkPartFileNotFound
+		}
+		return
```

Comment:
What about the paths to this return that don't set the response?

(I realize this is just code extracted, but it still seems strange)

Reply:
By default, the result is unknown, which is zero. We do not act one unknown results

Reply:
ok. But it never reads through the part - that is done in other checks, correct? 

If so, I am fine with the deadline and you can just resolve this.

Reply:
ok.